File: content/snippets/go/interview_lcci/16/11_diving_board.go
package main

import "fmt"

func divingBoard(shorter int, longer int, k int) []int {
	if k == 0 {
		return []int{}
	}
	if shorter == longer {
		return []int{shorter * k}
	}
	ans := make([]int, 0, k+1)
	for i := 0; i <= k; i++ {
		ans = append(ans, shorter*(k-i)+longer*i)
	}
	return ans
}

func main() {
	fmt.Println(divingBoard(1, 2, 3))
}

File: content/snippets/typescript/interview_lcci/16/11_diving_board.ts
function divingBoard(shorter: number, longer: number, k: number): number[] {
    if (k === 0) return [];
    if (shorter === longer) return [shorter * k];
    const ans: number[] = [];
    for (let i = 0; i <= k; i++) {
        ans.push(shorter * (k - i) + longer * i);
    }
    return ans;
}

console.log(divingBoard(1, 2, 3));
